server/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -750,21 +750,32 @@ public final <T extends TransportResponse> void sendRequest(
         final TransportRequestOptions options,
         TransportResponseHandler<T> handler
     ) {
-        final Transport.Connection connection;
+        final Transport.Connection connection = getConnectionOrFail(node, action, handler);
+        if (connection != null) {
+            sendRequest(connection, action, request, options, handler);
+        }
+    }
+
+    /**
+     * Get connection to the given {@code node} if possible and fail {@code handler} otherwise.
+     *
+     * @return connection if one could be found to {@code node} or null otherwise
+     */
+    @Nullable
+    private Transport.Connection getConnectionOrFail(DiscoveryNode node, String action, TransportResponseHandler<?> handler) {
         try {
-            connection = getConnection(node);
+            return getConnection(node);
```

Comment:
Paranoia: if some implementation of `getConnection()` itself returns `null` then we leak the handler here. Can we check for that case just to be sure?

Reply:
makes sense, I pushed [9eec9a6](https://github.com/elastic/elasticsearch/pull/98321/commits/9eec9a6e70eeb05ef22dc5871a50bf17d1e1ea74)